crates/shared-backup-db/src/postgres_connector.rs

```diff
@@ -1,82 +1,94 @@
+        let pool = Pool::builder(mgr)
+            .max_size(pg_config.max_pool_size().unwrap_or(16))
+            .build()
+            .unwrap();
+        let client = pool.get().await?;
         // create new db
         let db_name = format!("citrea{}", get_db_extension());
         let _ = client
             .batch_execute(&format!("CREATE DATABASE {};", db_name.clone()))
             .await;
-        drop(client);
+        drop(pool);
         //connect to new db
-        let (client, connection) = tokio_postgres::connect(
-            pg_config
-                .parse_to_connection_string_with_db(db_name)
-                .as_str(),
-            NoTls,
-        )
-        .await?;
-        tokio::spawn(async move {
-            if let Err(e) = connection.await {
-                eprintln!("connection error: {}", e);
-            }
-        });
+        cfg.dbname(&db_name);
+        let mgr = Manager::from_config(cfg, NoTls, mgr_config);
+        let pool = Pool::builder(mgr)
+            .max_size(pg_config.max_pool_size().unwrap_or(16))
+            .build()
+            .unwrap();
+        let client = pool.get().await?;
 
         // create tables
         client
             .batch_execute(SEQUENCER_COMMITMENT_TABLE_CREATE_QUERY)
             .await?;
-        let db_client = Self { client };
+        client.batch_execute(MEMPOOL_TXS_TABLE_CREATE_QUERY).await?;
+        let db_client = Self { client: pool };
 
         let _ = db_client.create_indexes().await;
 
         Ok(db_client)
     }
 
-    pub async fn create_indexes(&self) -> Result<(), Error> {
-        self.client.batch_execute(INDEX_L1_END_HEIGHT).await?;
-        self.client.batch_execute(INDEX_L1_END_HASH).await?;
-        self.client.batch_execute(INDEX_L2_END_HEIGHT).await?;
+    pub async fn client(&self) -> Result<Object, PoolError> {
+        Ok(self.client.get().await?)
+    }
+
+    pub async fn create_indexes(&self) -> Result<(), PoolError> {
+        let client = self.client().await.map_err(|e| e)?;
+        client.batch_execute(INDEX_L1_END_HEIGHT).await?;
+        client.batch_execute(INDEX_L1_END_HASH).await?;
+        client.batch_execute(INDEX_L2_END_HEIGHT).await?;
+        // client.batch_execute(INDEX_MEMPOOL_TXS).await?;
         Ok(())
     }
 
     #[cfg(feature = "test-utils")]
-    pub async fn new_test_client() -> Result<Self, Error> {
-        let pg_config = SharedBackupDbConfig::default();
-
-        let (client, connection) =
-            tokio_postgres::connect(pg_config.parse_to_connection_string().as_str(), NoTls).await?;
-        tokio::spawn(async move {
-            if let Err(e) = connection.await {
-                eprintln!("connection error: {}", e);
-            }
-        });
+    pub async fn new_test_client() -> Result<Self, PoolError> {
+        let mut cfg = PgConfig::new();
+        cfg.host("localhost")
+            .port(5432)
+            .user("postgres")
+            .password("password")
```

Comment:
It used to be postgres

Reply:
Yeah it should be don't know how I did that :D Thanks